### PR TITLE
CONTRIBUTING.md: troubleshooting build mini-guide

### PR DESCRIPTION
**Pull request by samzilverberg**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -33,6 +33,9 @@ make test
 make test-integration
 ```
 
+if you are having trouble building check out troubleshooting area below.
```
Comment (dominikschulz) [inline CONTRIBUTING.md]: Capitalization and informal speech.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -43,3 +46,39 @@ make test-integration
 * Commits should be as small as possible, while ensuring that each commit is correct independently.
 
 * Add tests relevant to the fixed bug or new feature.
+
+
+## Troubleshooting build
+
+### docker approach
```
Comment (dominikschulz) [inline CONTRIBUTING.md]: Capitalization

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -43,3 +46,39 @@ make test-integration
 * Commits should be as small as possible, while ensuring that each commit is correct independently.
 
 * Add tests relevant to the fixed bug or new feature.
+
+
+## Troubleshooting build
```
Comment (dominikschulz) [inline CONTRIBUTING.md]: I'd just call this `Troubleshooting`

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -43,3 +46,39 @@ make test-integration
...
+
+### docker approach
+you can build in a docker container by going to root dir of gopass project and:
```
Comment (dominikschulz) [inline CONTRIBUTING.md]: Capitalization, missing article confusing use of project. (`the root dir of your gopass checkout`).

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -43,3 +46,39 @@ make test-integration
...
+```
+docker run --rm -v "$PWD":/go/src/github.com/justwatchcom/gopass -w /go/src/github.com/justwatchcom/gopass golang:stretch go build -v
+```
+
+replace `go build -v` in above command with `make test` or anything else you'd like to run inside docker.
```
Comment (dominikschulz) [inline CONTRIBUTING.md]: Capitalization. Also I'd replace `anything else` with something along the line of `any other command`.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -43,3 +46,39 @@ make test-integration
...
+
+you can also get an interactive shell inside the container via:
```
Comment (dominikschulz) [inline CONTRIBUTING.md]: Capitalization

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -43,3 +46,39 @@ make test-integration
...
+```
+docker run -it -v "$PWD":/go/src/github.com/justwatchcom/gopass -w /go/src/github.com/justwatchcom/gopass golang:stretch bash
+```
+in which you can run `go build`, `make test` or whatever you'd like
```
Comment (dominikschulz) [inline CONTRIBUTING.md]: Capitalization, punctuation.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -43,3 +46,39 @@ make test-integration
...
+
+### non-docker
+
+make sure $GOPATH and $GOROOT are set correctly.
```
Comment (dominikschulz) [inline CONTRIBUTING.md]: `Please make sure ...`

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -43,3 +46,39 @@ make test-integration
...
+
+replace `go build -v` in above command with `make test` or anything else you'd like to run inside docker.
+
+you can also get an interactive shell inside the container via:
+```
+docker run -it -v "$PWD":/go/src/github.com/justwatchcom/gopass -w /go/src/github.com/justwatchcom/gopass golang:stretch bash
+```
+in which you can run `go build`, `make test` or whatever you'd like
+
+### non-docker
```
Comment (dominikschulz) [inline CONTRIBUTING.md]: Maybe something like `Non-Docker Approach`?

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -43,3 +46,39 @@ make test-integration
...
+
+make sure $GOPATH and $GOROOT are set correctly.
+$GOROOT should be the go dir where VERSION file is, and which has bin/ pkg/ and src/ as sub dirs.
```
Comment (dominikschulz) [inline CONTRIBUTING.md]: This line doesn't make much sense to me.

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -43,3 +46,39 @@ make test-integration
...
+$GOPATH is roughly where your go projects src code is.
```
Comment (dominikschulz) [inline CONTRIBUTING.md]: Maybe link to https://github.com/golang/go/wiki/GOPATH ?

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -43,3 +46,39 @@ make test-integration
...
+`go env` will show you the current dirs used.
```
Comment (dominikschulz) [inline CONTRIBUTING.md]: Rather: `You can use go env to show the directories currently used.`

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -43,3 +46,39 @@ make test-integration
...
+
+heres a quick start:
```
Comment (dominikschulz) [inline CONTRIBUTING.md]: Too informal

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -43,3 +46,39 @@ make test-integration
...
+- `mkdir $HOME/go-workspace`
```
Comment (dominikschulz) [inline CONTRIBUTING.md]: I wouldn't suggest to set `GOPATH=$HOME/go-workspace`, we should recommend the default of `GOPATH=$HOME/go`.

File: CONTRIBUTING.md [hunk 15]
```diff
@@ -43,3 +46,39 @@ make test-integration
...
+- `export GOPATH=$HOME/go-workspace`
+- `mkdir $GOPATH/src`
+- `mkdir -p $GOPATH/src/github.com/justwatchcom/`
```
Comment (dominikschulz) [inline CONTRIBUTING.md]: Actually I'd recommend to just run `go get -u github.com/justwatchcom/gopass`.

File: CONTRIBUTING.md [hunk 16]
```diff
@@ -43,3 +46,39 @@ make test-integration
...
+- clone gopath project the the above dir (so its in $GOPATH/src/github.com/justwatchcom/golang)
+- making sure $GOROOT points to /usr/local/opt/go/libexec/ (installed golang via brew)
```
Comment (dominikschulz) [inline CONTRIBUTING.md]: This is too MacOS specific.

**Comment by samzilverberg** [comment]
@dominikschulz I made a lot of changes based on your comments.
If you want to change some more I suggest you just squash+merge then make the changes you want directly on master. 
Otherwise I feel it would jsut become a ping-pong comments about a small info file that would probably be a waste of our for both of us ;)

**Pull request merged**